pkg/urbit/vere/pier.c

```diff
@@ -60,13 +98,32 @@ _pier_work_send(u3_work* wok_u)
   {
     u3_ovum* egg_u;
     u3_noun    ovo;
+    u3_pico* pic_u;
 
-    while (  len_w-- && car_u && (egg_u = u3_auto_next(car_u, &ovo)) ) {
+    while ( len_w && car_u && (egg_u = u3_auto_next(car_u, &ovo)) ) {
+      len_w--;
```

Comment:
Initial test on ~nus resulted in the same old unresponsive ship, seemingly exclusively busy with packet forwarding. After this specific change, the ship is responsive again.

The commit message doesn't mention this, but I now think this was just integer underflowing in the `len_w == 0` case, leading to enqueuing a ton of scries further down in this function. (Of course, this also fixes handling of the "no next ovum" case, but that was less terribly wrong.)

Reply:
Good catch! I got a little too cute with the one-liner ...